pkgs/test/cuda/cuda-samples/generic.nix

```diff
@@ -0,0 +1,44 @@
+{ lib, stdenv, fetchFromGitHub
+, pkg-config
+, sha256, cudatoolkit
+}:
+
+let
+  pname = "cuda-samples";
+  version = lib.versions.majorMinor cudatoolkit.version;
+in
+
+stdenv.mkDerivation {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    owner = "NVIDIA";
+    repo = pname;
+    rev = "v${version}";
+    inherit sha256;
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+
+  propagatedBuildInputs = [ cudatoolkit ];
+
+  enableParallelBuilding = true;
+
+  preConfigure = ''
+    export CUDA_PATH=${cudatoolkit}
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm755 -t $out/bin bin/${stdenv.hostPlatform.parsed.cpu.name}/${stdenv.hostPlatform.parsed.kernel.name}/release/*
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Samples for CUDA Developers which demonstrates features in CUDA Toolkit.";
+    # CUDA itself is proprietary, but these sample apps are not.
+    license = lib.licenses.bsd3;
```

Comment:
I think it wouldn't hurt if you added yourself as a maintainer.

Reply:
I am not sure who is going to end up maintaining this yet, so I'm going to mark it as WIP.